ggshield/scan/docker.py

```diff
@@ -336,13 +348,15 @@ def docker_scan_archive(
             if file_count == 0:
                 continue
             print()
-            display_heading(f"Scanning layer {info.get_id()}")
-            with RichSecretScannerUI(file_count) as ui:
-                results.extend(
-                    scanner.scan(
-                        layer.files,
-                        scanner_ui=ui,
-                    )
-                )
+            layer_id = info.get_id()
+            if layer_id in layer_id_cache:
+                display_heading(f"Skipping layer {layer_id}: already scanned")
```

Comment:
It may be clearer to add that the layer did not contain any secret.

Reply:
Also, this can become a bit verbose, the caching logic is somehow an implementation detail... shouldn't we not print this message by default ?

Reply:
I think it is better to be explicit here because otherwise the user is going to be surprised by the very different duration of the scan for cached layers.